src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -142,13 +142,22 @@ export class SearchSource {
    */
   setField<K extends keyof SearchSourceFields>(field: K, value: SearchSourceFields[K]) {
     if (value == null) {
-      delete this.fields[field];
+      return this.removeField(field);
     } else {
       this.fields[field] = value;
     }
     return this;
   }
 
+  /**
+   * remove field
+   * @param field: field name
+   */
+  removeField<K extends keyof SearchSourceFields>(field: K) {
+    delete this.fields[field];
+    return this;
+  }
```

Comment:
Why did I add this, to prevent TypeScript confusion, because when I used 'setFields(key, null)' to remove it, I got depending what I was trying to remove TypeScript errors

Reply:
Would be nice if we could add a quick unit test for the new method

Reply:
Absolutely! I've added a jest test.